File: Runtime/CoreLib/JsErrorException.cs
using System.Runtime.CompilerServices;

namespace System {
	[Imported(ObeysTypeSystem = true)]
	[ScriptNamespace("ss")]
	public class JsErrorException : Exception {
		public JsErrorException(Error error) {
		}

		public JsErrorException(Error error, string message) {
		}

		public JsErrorException(Error error, string message, Exception innerException) {
		}

		[IntrinsicProperty]
		public Error Error { get { return null; } }

		/// <summary>
		/// The stack of the originating Javascript Error, if available.
		/// </summary>
		public override string Stack {
			get { return null; }
		}
	}
}

File: Runtime/CoreLib.Script/JsErrorException.js
////////////////////////////////////////////////////////////////////////////////
// JsErrorException

var ss_JsErrorException = ss.JsErrorException = ss.mkType(ss, 'ss.JsErrorException', function#? DEBUG JsErrorException$##(error, message, innerException) {
		ss_Exception.call(this, message || error.message, innerException);
		this.error = error;
	},
	{
		get_stack: function#? DEBUG Exception$get_stack##() {
			return this.error.stack;
		}
	}
);

ss.initClass(ss_JsErrorException, ss_Exception);
